Code/Engine/RendererVulkan/CommandEncoder/Implementation/CommandEncoderImplVulkan.cpp

```diff
@@ -491,23 +516,59 @@ void ezGALCommandEncoderImplVulkan::BeginRendering(vk::CommandBuffer& commandBuf
 {
   m_pCommandBuffer = &commandBuffer;
 
-  vk::RenderPassCreateInfo renderPassCreateInfo;
-  renderPassCreateInfo.attachmentCount = 0;
-  // TODO: fill render pass create info
-  // TODO: caching
+  ezResourceCacheVulkan::RenderPassDesc renderPassDesc;
+  ezResourceCacheVulkan::GetRenderPassDesc(renderingSetup, renderPassDesc);
+  vk::RenderPass renderPass = ezResourceCacheVulkan::RequestRenderPass(renderPassDesc);
 
-  vk::RenderPass renderPass = m_vkDevice.createRenderPass(renderPassCreateInfo);
+  ezResourceCacheVulkan::FramebufferDesc framebufferDesc;
+  ezResourceCacheVulkan::GetFrameBufferDesc(renderPass, renderingSetup, framebufferDesc);
+  vk::Framebuffer frameBuffer = ezResourceCacheVulkan::RequestFrameBuffer(framebufferDesc);
 
   vk::RenderPassBeginInfo renderPassBeginInfo;
-  renderPassBeginInfo.renderPass = renderPass;
+  {
+    renderPassBeginInfo.renderPass = renderPass;
+    renderPassBeginInfo.framebuffer = frameBuffer;
+    renderPassBeginInfo.renderArea.offset.setX(0).setY(0);
+    renderPassBeginInfo.renderArea.extent.setHeight(framebufferDesc.height).setWidth(framebufferDesc.width);
+
+    ezHybridArray<vk::ClearValue, EZ_GAL_MAX_RENDERTARGET_COUNT> clearValues;
+
+    const bool bHasDepth = !renderingSetup.m_RenderTargetSetup.GetDepthStencilTarget().IsInvalidated();
+    const ezUInt32 uiColorCount = renderingSetup.m_RenderTargetSetup.HasRenderTargets() ? renderingSetup.m_RenderTargetSetup.GetMaxRenderTargetIndex() + 1 : 0;
+    if (bHasDepth)
+    {
+      vk::ClearValue& depthClear = clearValues.ExpandAndGetRef();
+      depthClear.depthStencil.setDepth(1.0f).setStencil(0);
+    }
+    for (ezUInt32 i = 0; i < uiColorCount; i++)
+    {
+      vk::ClearValue& colorClear = clearValues.ExpandAndGetRef();
+      ezColor col = renderingSetup.m_ClearColor;
+      colorClear.color.setFloat32({col.r, col.g, col.b, col.a});
+    }
+
+    renderPassBeginInfo.clearValueCount = clearValues.GetCount();
+    renderPassBeginInfo.pClearValues = clearValues.GetData();
+  }
 
   m_pCommandBuffer->beginRenderPass(renderPassBeginInfo, vk::SubpassContents::eInline);
+
+  vk::Viewport viewport;
```

Comment:
Viewport and scissor setup are separate commands, so they should not be set here.

Reply:
Yep, removed.